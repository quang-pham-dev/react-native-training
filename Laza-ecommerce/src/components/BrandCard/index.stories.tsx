import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import CenterView from '../../../storybook/stories/CenterView';
import BrandCard from './index';
import { brand } from '__mocks__/dataMock/brands';

storiesOf('Brand Card', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Default', () => <BrandCard brand={brand} onPressBrandCard={action('clicked')} />);
