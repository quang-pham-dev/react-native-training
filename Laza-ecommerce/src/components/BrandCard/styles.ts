import { StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Metrics from 'themes/Metrics';

const styles = StyleSheet.create({
  brandCardContainer: {
    justifyContent: 'center',
  },
  brandCardWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 10,
    height: 50,
    backgroundColor: Colors.secondaryBackground,
    marginRight: Metrics.margin.sm,
    paddingRight: Metrics.padding.sm,
  },
  brandLogoWrapper: {
    width: 40,
    height: 40,
    padding: Metrics.padding.xs,
    borderRadius: 10,
    marginLeft: Metrics.margin.xsm,
    marginRight: Metrics.margin.sm,
    marginVertical: Metrics.margin.xsm,
    backgroundColor: Colors.primaryBackground,
  },
  brandLogo: {
    width: '100%',
    height: '100%',
    resizeMode: 'contain',
  },
});

export default styles;
