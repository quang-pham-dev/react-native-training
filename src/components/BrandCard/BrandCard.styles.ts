import styled from 'styled-components/native'

// Libs
import {s} from 'react-native-size-matters/extend'

// Themes
import {Colors, Metrics} from '@themes'

export const BrandCardWrapperStyled = styled.TouchableOpacity`
  height: ${s(50)}px;
  margin-right: ${Metrics.margin.small}px;
  background-color: ${Colors.palette.lightGray};
  border-radius: ${Metrics.borderRadius.base}px;
  padding-right: ${Metrics.padding.medium}px;
`
export const ImageWrapperStyled = styled.View`
  height: ${s(40)}px;
  width: ${s(40)}px;
  justify-content: center;
  align-items: center;
  padding-vertical: ${Metrics.padding.xs}px;
  padding-horizontal: ${Metrics.padding.xs}px;
  margin-left: ${Metrics.margin.tiny}px;
  margin-right: ${Metrics.margin.small}px;
  margin-vertical: ${Metrics.margin.tiny}px;
  border-radius: ${Metrics.borderRadius.base}px;
  background-color: ${Colors.background};
`
