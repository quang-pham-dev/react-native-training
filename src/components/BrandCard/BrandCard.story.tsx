import React from 'react'

// Storybook
import {action} from '@storybook/addon-actions'
import {storiesOf} from '@storybook/react-native'
import {StoryScreen, Story, UseCase} from '../../../storybook/views'

// Components
import BrandCard from './index'

// Constants
import {brand} from '@constants/dataMock'

declare let module: NodeModule

storiesOf('Brand Card', module)
  .addDecorator(fn => <StoryScreen>{fn()}</StoryScreen>)
  .add('Default', () => (
    <Story>
      <UseCase text="default">
        <BrandCard brand={brand} onPressBrandCard={action('clicked')} />
      </UseCase>
    </Story>
  ))
