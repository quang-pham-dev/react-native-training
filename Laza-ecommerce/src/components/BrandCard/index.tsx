import React, { memo, useCallback } from 'react';
import { Image, TouchableOpacity, View } from 'react-native';

// LIBS
import isEqual from 'react-fast-compare';

// Components
import Title from 'components/Title';

// Types
import { IBrandCardProps } from 'types/models/Brands';

// Styles
import styles from './styles';

const BrandCard = ({ brand, onPressBrandCard, brandCardStyles }: IBrandCardProps) => {
  // handle action press card brand
  const handlePressBrandCard = useCallback(() => {
    onPressBrandCard(brand.id);
  }, [brand.id, onPressBrandCard]);

  return (
    <TouchableOpacity
      testID='brandCard'
      onPress={handlePressBrandCard}
      style={[styles.brandCardContainer, brandCardStyles]}>
      <View style={[styles.brandCardWrapper]}>
        <View style={styles.brandLogoWrapper}>
          <Image style={styles.brandLogo} source={{ uri: brand.logoUrl }} />
        </View>
        <Title titleName={brand.name} />
      </View>
    </TouchableOpacity>
  );
};

export default memo(BrandCard, isEqual);
