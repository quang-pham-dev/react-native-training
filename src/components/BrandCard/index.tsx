import React, {memo, useCallback} from 'react'

// LIBS
import isEqual from 'react-fast-compare'
import FastImage from 'react-native-fast-image'
import {s} from 'react-native-size-matters/extend'

// Types
import {BrandCardProps} from '@model-types'

// Styles
import {BrandCardWrapperStyled, ImageWrapperStyled} from './BrandCard.styles'
import PStyled from '@components/Paragraph/P.styles'
import {Image} from '@components/Image/Image.styles'
import FlexStyled from '@components/Flex/Flex.styles'

// Constants
import {PARAGRAPH_TYPE} from '@constants'

const BrandCard = ({brand, onPressBrandCard}: BrandCardProps) => {
  const {id, name, url} = brand || {}
  // handle action press card brand
  const handlePressBrandCard = useCallback(() => {
    onPressBrandCard(id)
  }, [id, onPressBrandCard])

  return (
    <BrandCardWrapperStyled onPress={handlePressBrandCard}>
      <FlexStyled.RowCenterHorizontal>
        <ImageWrapperStyled>
          <Image.Normal
            alignSelf="center"
            height={s(40)}
            width={s(40)}
            source={{
              uri: url,
              priority: FastImage.priority.normal,
            }}
            resizeMode={FastImage.resizeMode.contain}
          />
        </ImageWrapperStyled>
        <PStyled.Base type={PARAGRAPH_TYPE.BRAND}>{name}</PStyled.Base>
      </FlexStyled.RowCenterHorizontal>
    </BrandCardWrapperStyled>
  )
}

export default memo(BrandCard, isEqual)
