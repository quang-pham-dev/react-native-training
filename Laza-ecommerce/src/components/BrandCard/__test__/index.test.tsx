import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import BrandCard from 'components/BrandCard';

const brandData = {
  id: '1',
  name: 'Nike',
  logoUrl: 'https://www.nike.com/a/content/dam/one_tap/one_tap_home/2019/07/nike-logo-white.png',
};

describe('Brand Card Component', () => {
  const navigate = jest.fn();

  const component = render(
    <BrandCard brand={brandData} handleNavigationBrandDetailScreen={navigate} />,
  );
  test('Should render correctly', () => {
    const result = component.toJSON();
    expect(result).toMatchSnapshot();
  });

  test('Should navigate is called', () => {
    const { getByTestId } = component;

    const button = getByTestId('brandCard');
    fireEvent.press(button);
    expect(navigate).toHaveBeenCalledTimes(1);
  });
});
