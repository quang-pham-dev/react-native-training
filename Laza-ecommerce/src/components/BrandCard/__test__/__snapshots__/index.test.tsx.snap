// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand Card Component Should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="brandCard"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F6FA",
          "borderRadius": 10,
          "flexDirection": "row",
          "height": 50,
          "marginRight": 10,
          "paddingRight": 10,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 10,
          "height": 40,
          "marginLeft": 5,
          "marginRight": 10,
          "marginVertical": 5,
          "padding": 8,
          "width": 40,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://www.nike.com/a/content/dam/one_tap/one_tap_home/2019/07/nike-logo-white.png",
          }
        }
        style={
          Object {
            "backgroundColor": "transparent",
            "height": "100%",
            "resizeMode": "contain",
            "width": "100%",
          }
        }
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#1D1E20",
              "fontSize": 17,
            },
            Object {
              "color": "#1D1E20",
              "fontFamily": "Inter_500Medium",
              "fontSize": 15,
              "lineHeight": 16.5,
            },
          ]
        }
      >
        Nike
      </Text>
    </View>
  </View>
</View>
`;
